### PR TITLE
T37035 CHROMEOS Integrate coral device type

### PR DESCRIPTION
**Pull request by JenySadadia**
rootfs-configs-chromeos.yaml: add chromiumos-coral
test-configs-chromeos.yaml: add chromiumos-coral
build_board.sh: apply patches specific to coral

### CODE REVIEW

**Comment by JenySadadia** [comment]
Tests run successfully after applying patches by @nuclearcat https://github.com/kernelci/kernelci-core/pull/1404 and https://github.com/kernelci/kernelci-core/pull/1381 :
https://lava.collabora.dev/scheduler/job/7164631
https://lava.collabora.dev/scheduler/job/7160998

**Review by nuclearcat — APPROVED** [review]
LGTM, flashed all corals and tested them on baseline tests, will wait results to appear on dashboard as well

**Comment by nuclearcat** [comment]
@JenySadadia can you rebase please?

**Comment by JenySadadia** [comment]
> @JenySadadia can you rebase please?

Done.

File: config/core/test-configs-chromeos.yaml [hunk 1]
```diff
@@ -314,6 +342,11 @@ test_configs:
       - cros-tast-perf
       - cros-tast-platform
       - cros-tast-video
+    
+  - device_type: asus-C523NA-A20057-coral_chromeos
+    test_plans:
+      - cros-baseline-fixed
+      - cros-flash
```
Comment (nuclearcat) [inline config/core/test-configs-chromeos.yaml:349]: Can we replace test plans by updated ones?
```
    test_plans:
      - cros-baseline
      - cros-baseline-fixed
      - cros-tast-kernel
      - cros-tast-perf
      - cros-tast-platform
      - cros-tast-video
```

**Review by nuclearcat — CHANGES_REQUESTED** [review]

File: config/core/test-configs-chromeos.yaml (hunk 1 above)
Comment (JenySadadia) [reply config/core/test-configs-chromeos.yaml:349]: Okay. Let me test them first.

Comment (JenySadadia) [reply config/core/test-configs-chromeos.yaml]: @nuclearcat  I have updated the configurations to use `chromebook-generic-params` and test plans.

File: config/core/test-configs-chromeos.yaml [hunk 2]
```diff
@@ -223,6 +223,30 @@ device_types:
...
 
+  asus-C523NA-A20057-coral_chromeos:
+    base_name: asus-C523NA-A20057-coral
+    mach: x86
+    arch: x86_64
+    boot_method: depthcharge
+    filters: *pineview-filter
+    params:
+      <<: *chromebook-generic-params
+      cros_flash_nfs:
```
Comment (nuclearcat) [inline config/core/test-configs-chromeos.yaml:234]: After using chromebook-generic-params you can remove crosh_flash_nfs section from device, as it is "derived" from chromebook-generic-params.
For example look to hp-11A-G6-EE-grunt_chromeos and hp-x360-12b-ca0010nr-n4020-octopus_chromeos

**Review by nuclearcat — CHANGES_REQUESTED** [review]

File: config/core/test-configs-chromeos.yaml (hunk 2 above)
Comment (JenySadadia) [reply config/core/test-configs-chromeos.yaml]: Updated.

**Comment by nuclearcat** [comment]
Chromebook being tested and appearing on dashboard:
https://storage.chromeos.kernelci.org/kernelci/chromeos-stable/chromeos-stable-20221004.0/x86_64/cros---chromeos-5.15-x86_64-chromeos-intel-pineview.flavour.config+x86-chromebook+CONFIG_MODULE_COMPRESS_GZIP=n/clang-14/lab-collabora/cros-tast-video-asus-C523NA-A20057-coral_chromeos.html
https://chromeos.kernelci.org/test/plan/id/633ba01bd7d190b9eba0e036/

**Review by nuclearcat — APPROVED** [review]
LGTM

**Review by mgalka — APPROVED** [review]
I think this is good enough to have it merged.

**Pull request merged**